demo/src/components/SrlComponent.js

```diff
@@ -34,6 +34,77 @@ const description = (
   </div>
 );
 
+function getStrIndex(words, wordIdx) {
+  if (wordIdx < 0) throw new Error(`Invalid word index: ${wordIdx}`);
+  return words.slice(0, wordIdx).join(' ').length;
+}
+
+function toHierplaneTrees(response) {
+  const text = response.words.join(' ');
+
+  // We create a tree for each verb
+  const trees = response.verbs.map(({ verb, tags }) => {
+    const verbTagIdx = tags.findIndex(tag => tag === 'B-V');
+    const start = getStrIndex(response.words, verbTagIdx);
+
+    const ignoredSpans = tags.reduce((children, tag, idx) => {
+      if (tag === 'O') {
+        const word = response.words[idx];
+        const start = getStrIndex(response.words, idx);
+        const end = start + word.length + 1;
+        const child = {
+          spanType: 'ignored',
+          start,
+          end
+        };
+        children.push(child);
+      }
+      return children;
+    }, []);
+
+    return {
+      text,
+      root: {
+        word: verb,
+        nodeType: 'V',
+        attributes: [ 'VERB' ],
+        spans: [{
+          start,
+          end: start + verb.length + 1,
+        }, ...ignoredSpans],
+        children: tags.reduce((children, tag, idx) => {
+          if (tag !== 'B-V' && tag.startsWith('B-')) {
+            const word = response.words[idx];
+            const link = tag.split('-').pop().replace(/\d+/g, '');
+            const start = getStrIndex(response.words, idx);
+            const newChild = {
+              word,
+              spans: [{
+                start,
+                end: start + word.length + 1
```

Comment:
What would be really nice is if we could make the `nodeType` and `attributes` different.
The SRL labels either contain `ARGM`, which modify the verb, or `ARG1-5` which are the core arguments. For the `ARGM`, the nodeType could be `modifier` and for the `ARG1-5` it could be `arg`, with the `attribute` containing the second half of the label, eg `TMP` or `DIS`.

One final point to be aware of is that some of the labels might contain `C-ARG1` or `R-ARG-1`, which mean that the label is continued or referred to, eg `(The place ARG1), beautifully though it was, (where R-ARG1) Mark lived`.

Reply:
Makes a ton of sense! I can easily make this happen.  I'll make the change and get a local demo up today so you can review it / play with it.

Reply:
Is the `PRP` suffix of value, or something I can disregard for this purpose?

Reply:
Ah, `PRP` is likely the "second half" -- I think I grok things now.

Would `R-` and `C-` both be appropriately labeled as a `reference`, or are they different types?

Reply:
R would be `reference` and C would be `continuation`, they are very rarely observed but can crop up so we should handle them.

PRP is actually the label, it is short hand for `purpose`, i.e why the verb is being done. I'm not exactly sure what you mean by second half, feel free to swing by and we can chat in person.

Reply:
🤘  Thx.  By the "second half" I mean what you mentioned here:

> with the attribute containing the second half of the label, eg TMP or DIS.

So I'm assuming `TMP`, `DIS` and `PRP` all serve similar purposes (pun intended!) :-D.  LMK if that's not the case.  Also I have an interactive version I can link to shortly.